dts/bindings/acpi/acpi-vbus.yaml

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for ACPI virtual bus
+
+include: base.yaml
+
+bus: acpi
```

Comment:
No ACPI is _not_ a hw bus. This is not compliant with DTS.

Reply:
Agree that ACPI is not a hardware bus, but it defined to standardizes OS interface with platform HW and FW which is not limited to power management but include planform configuration such as interrupt routing, DMA, device MMIO resources, device enumeration and many more. ACPI driver itself have registers sets and ISR to mange its run time functionalities.  So in implementation point of view, it is more like a driver which deals with HW and its associated interface. For more details of ACPI: https://uefi.org/htmlspecs/ACPI_Spec_6_4_html/01_Introduction/Introduction.html
 
I believe  the question which is debatable here is DTS vs ACPI, that is whether we can completely move to ACPI base model for drivers used in platforms like x86 which include device enumeration apart from resource enumerations. But currently Zephyr moving to DTS base including application interface and most of the test apps started using dts base interface such as DEVICE_DT_GET instead of device_get_binding().  So removing DTS support for some platform is not viable currently without complicate these test apps.

Following option we may need to think:
- Option 1:  Continue the DTS base device enumeration (discover devices in a platform) for ACPI devices just like currently how PCI device enumerated and will be consistent with existing Zephyr driver's application/usage model, but all other MMIO/IRQ resource enumeration will be via ACPI like PCI devices. This need some bus node to distinguish the driver's parent node in DTS like pcie. Currently I follow this option and provide a virtual bus node name "acpi".

- Option 2: Move to a complete ACPI base model for all non PCI devices in x86 platforms which include device enumerations and without any DTS support. Here PCI driver interface also need to move completely on PCI based device enumeration (Device discovery)

Reply:
removed ACPI node. In order to distinguish  whether a driver use ACPI interface or not, one can use DT_NODE_HAS_PROP(n, acpi_uid)). Please refer PR(https://github.com/zephyrproject-rtos/zephyr/pull/62993) for more details of the usage.

@galak could you please have look on this PR as well as PR(https://github.com/zephyrproject-rtos/zephyr/pull/62993) ?

Reply:
@tbursztyka @dcpleung could you please review the PR ?